test_conformance/half/Test_vLoadHalf.cpp

```diff
@@ -35,6 +36,8 @@ int Test_vLoadHalf_private( cl_device_id device, bool aligned )
     uint64_t min_time[kVectorSizeCount+kStrangeVectorSizeCount] = {0};
     size_t q;
 
+    if (!is_extension_available(device, "cl_khr_fp16")) return 0;
```

Comment:
Could you elaborate on why we need these checks for `cl_khr_fp16`? My understanding is that `vload{a}_halfn` and `vstore{a}_halfn` are part of the core OpenCL-C spec rather than the fp16 extension, and so need support even if a device doesn't support this extension.

Reply:
seems like you are right. It seems like I was simply hitting an clang bug then not providing those functions unless the extension was enabled.